modules/default/utils.js

```diff
@@ -0,0 +1,146 @@
+			corsUrl = `${corsUrl}&expectedheaders=${expectedResponseHeadersString}`;
+		} else if (expectedResponseHeadersString) {
+			corsUrl = `${corsUrl}expectedheaders=${expectedResponseHeadersString}`;
+		}
+
+		if (requestHeaderString || expectedResponseHeadersString) {
+			return `${corsUrl}&url=${url}`;
+		}
+		return `${corsUrl}url=${url}`;
+	}
+};
+
+/**
+ * Gets the part of the CORS URL that represents the HTTP headers to send.
+ *
+ * @param {Array.<{name: string, value:string}>} requestHeaders the HTTP headers to send
+ * @returns {string} to be used as request-headers component in CORS URL.
+ */
+const getRequestHeaderString = function (requestHeaders) {
+	let requestHeaderString = "";
+	if (requestHeaders) {
+		for (const header of requestHeaders) {
+			if (requestHeaderString.length === 0) {
+				requestHeaderString = `${header.name}:${encodeURIComponent(header.value)}`;
+			} else {
+				requestHeaderString = `${requestHeaderString},${header.name}:${encodeURIComponent(header.value)}`;
+			}
+		}
+		return requestHeaderString;
+	}
+	return undefined;
+};
+
+/**
+ * Gets headers and values to attatch to the web request.
+ *
+ * @param {Array.<{name: string, value:string}>} requestHeaders the HTTP headers to send
+ * @returns {object} An object specifying name and value of the headers.
+ */
+const getHeadersToSend = (requestHeaders) => {
+	const headersToSend = {};
+	if (requestHeaders) {
+		for (const header of requestHeaders) {
+			headersToSend[header.name] = header.value;
+		}
+	}
+
+	return headersToSend;
+};
+
+/**
+ * Gets the part of the CORS URL that represents the expected HTTP headers to recieve.
+ *
+ * @param {Array.<string>} expectedResponseHeaders the expected HTTP headers to recieve
+ * @returns {string} to be used as the expected HTTP-headers component in CORS URL.
+ */
+const getExpectedResponseHeadersString = function (expectedResponseHeaders) {
+	let expectedResponseHeadersString = "";
+	if (expectedResponseHeaders) {
+		for (const header of expectedResponseHeaders) {
+			if (expectedResponseHeadersString.length === 0) {
+				expectedResponseHeadersString = `${header}`;
+			} else {
+				expectedResponseHeadersString = `${expectedResponseHeadersString},${header}`;
+			}
+		}
+		return expectedResponseHeaders;
+	}
+	return undefined;
+};
+
+/**
+ * Gets the values for the expected headers from the response.
+ *
+ * @param {Array.<string>} expectedResponseHeaders the expected HTTP headers to recieve
+ * @param {Response} response the HTTP response
+ * @returns {string} to be used as the expected HTTP-headers component in CORS URL.
+ */
+const getHeadersFromResponse = (expectedResponseHeaders, response) => {
+	const responseHeaders = [];
+
+	if (expectedResponseHeaders) {
+		for (const header of expectedResponseHeaders) {
+			const headerValue = response.headers.get(header);
+			responseHeaders.push({ name: header, value: headerValue });
+		}
+	}
+
+	return responseHeaders;
+};
+
+module.exports = {
```

Comment:
This isnt recognized by the brwoser and leads to 
"Uncaught ReferenceError: module is not defined
    at utils.js:144:1
    "

Reply:
strange, "works on my machine"... I'll fix it 👍

Reply:
How about now? I used the `if (typeof module !== "undefined")`-check used in several other parts of the code.